### PR TITLE
support python apps with no package manager + Procfile

### LINKED ISSUE #218: Restructure: Add an order-group that enables running python without any package managers
### Background
See [RFC 0001](https://github.com/paketo-community/python/blob/main/rfcs/0001-restructure.md) that outlines the proposed restructure of python family of buildpacks.  In order to conform to the RFC, a new order-group should be added to the metacnb that enables running python without any package managers.

### Proposal

* Add the following order-group to [python](https://github.com/paketo-community/python/blob/main/buildpack.toml) as the last item in the order:
```
[[order]]

  [[order.group]]
    id = "paketo-community/cpython"

  [[order.group]]
    id = "paketo-community/python-start"

  [[order.group]]
    id = "paketo-buildpacks/procfile"
    optional = true
```

* Add relevant integration tests

### PR DESCRIPTION
**Pull request by fg-j**
Resolves #218

Signed-off-by: Emily Johnson <emjohnson@vmware.com>

Note: Why isn't the automatic "so, you opened a pull request" template showing up in here?

### CODE REVIEW

**Comment by fg-j** [comment]
Note: The `python-start` buildpack sets the start command to `python`, which starts the python interpreter. So if you `docker run -it <app image>`, you find yourself inside the python interpreter. But!

If you `docker run -it <app image> server.py` (which we would [expect to be equivalent to running `python server.py`](https://github.com/buildpacks/rfcs/blob/main/text/0045-launcher-arguments.md#providing-additional-arguments-at-runtime)), you _also_ are dropped into the python interpreter, instead of a container that's running the file `server.py`.

This isn't necessarily bad/wrong behaviour, but it's surprising given the upstream [launcher arguments RFC](https://github.com/buildpacks/rfcs/blob/main/text/0045-launcher-arguments.md) and is probably worth documenting/investigating a bit more.

**Comment by ryanmoran** [comment]
I think we need to make sure that the buildpack API we use is a recent one: https://github.com/paketo-community/python-start/blob/13410214ad03ce112eef9a45040440507d8d35d1/buildpack.toml#L1

File: integration/no_package_manager_test.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+			image, logs, err = pack.WithNoColor().Build.
+				WithBuildpacks(pythonBuildpack).
+				WithPullPolicy("never").
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred(), logs.String())
+
+			Expect(logs).To(ContainLines(ContainSubstring("CPython Buildpack")))
+			Expect(logs).To(ContainLines(ContainSubstring("Python Start Buildpack")))
+
+			container, err = docker.Container.Run.
```
Comment (arjun024) [inline integration/no_package_manager_test.go:68]: My only quibble is that I'd expect this test to create an image that launches a repl on run, but the test is overriding the launch cmd.  Could this check for the repl instead?

**Review by arjun024 — CHANGES_REQUESTED** [review]

File: integration/no_package_manager_test.go (hunk 1 above)
Comment (fg-j) [reply integration/no_package_manager_test.go:68]: We figured that the `python-start` integration tests would be checking for that start command specifically (and that these tests are more about ensuring that the right order group is activated via build logs).

**Comment by fg-j** [comment]
I believe the linked RFC about launcher args has been implemented since [lifecycle 0.9.0](https://github.com/buildpacks/lifecycle/releases/tag/v0.9.0), so it's available in [pack 0.13.0 and later](https://github.com/buildpacks/pack/releases/tag/v0.13.1).

File: integration/no_package_manager_test.go (hunk 1 above)
Comment (arjun024) [reply integration/no_package_manager_test.go:68]: I believe paketo language families do end testing on the resultant image (e.g make sure ca-certs loaded in the final image). Not a strong opinion - ready to merge if you think this is fine.

Comment (fg-j) [reply integration/no_package_manager_test.go:68]: After bumping API version to 0.4, we'll be able to test that the `python-start` sets the right start command *and* that we can append a target file as an arg, which will be nice end-to-end testing.

**Comment by fg-j** [comment]
We'll need to merge the PR linked above in order to have this one go green.

File: integration/no_package_manager_test.go (hunk 1 above)
Comment (arjun024) [reply integration/no_package_manager_test.go:68]: Thanks

**Comment by arjun024** [comment]
@fg-j @emmjohnson You might have to update package.toml for the tests to pass. Upstream buildpacks with new api versions are released.

**Review by arjun024 — APPROVED** [review]

**Pull request merged**